### PR TITLE
add readme to the player npm package

### PR DESCRIPTION
**Pull request by Caleb335**
Hi @JonnyBurger,

Kindly take review the change that I made to the player npm package and let me know what you think. 

I'll also get started with the GIF package

### CODE REVIEW

File: packages/player/README.md [hunk 1]
```diff
@@ -4,5 +4,76 @@
...
+};
+```
+
+## The props in Player and their roles
+
+- `component`: the component prop takes in a component, a javascript function or a class when it is exported from its module
+- `durationInFrames`: represents the amount of time a frame will take to render
+- `compositionHeight`: represents the height that video component will occupy in the viewport
+- `compositionWidth`: represents the width that video component will occupy in the viewport
+- `fps`: **Frame Per Seconds** respresents the time, in seconds, that any number of frmae will be rendered in the video. The snippet above has 30 `fps`, which means the video will render 30 frames in a second throughout the whole duration of the video
```
Comment (anush008) [inline packages/player/README.md]: There are a few typos here, namely *frmae, *respresents. The description of the `fps`  prop is incorrect.

File: packages/player/README.md [hunk 2]
```diff
@@ -4,5 +4,76 @@
 [![NPM Downloads](https://img.shields.io/npm/dm/@remotion/player.svg?style=flat)](https://npmcharts.com/compare/@remotion/player?minimal=true)
 [![Install Size](https://packagephobia.now.sh/badge?p=@remotion/player)](https://packagephobia.now.sh/result?p=@remotion/player)
 
-This package allows you to include a Remotion video in your React app.
-Full documentation can be found under http://remotion.dev/docs/player.
+To use this package, we'll be asssuming you already have the required setup to use remotion to create awesome videos with React. The dependecies that remotion requires you have pre-installed on your machine are Node.js and FFMPEG. you can take a look at this [guide](https://github.com/adaptlearning/adapt_authoring/wiki/Installing-FFmpeg) on how to get FFMPEG on your machine.
+
+
+The `@remotion/player` package allows you to include a video powered by Remotion in a React application. To gest started with using this package, you need to have this package installed as a dependency in your project.
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
The `@remotion/player` package allows you to include a video powered by Remotion in a React application. To get started with using this package, you need to have this package installed as a dependency in your project. 
```

File: packages/player/README.md [hunk 3]
```diff
@@ -4,5 +4,76 @@
 [![NPM Downloads](https://img.shields.io/npm/dm/@remotion/player.svg?style=flat)](https://npmcharts.com/compare/@remotion/player?minimal=true)
 [![Install Size](https://packagephobia.now.sh/badge?p=@remotion/player)](https://packagephobia.now.sh/result?p=@remotion/player)
 
-This package allows you to include a Remotion video in your React app.
-Full documentation can be found under http://remotion.dev/docs/player.
+To use this package, we'll be asssuming you already have the required setup to use remotion to create awesome videos with React. The dependecies that remotion requires you have pre-installed on your machine are Node.js and FFMPEG. you can take a look at this [guide](https://github.com/adaptlearning/adapt_authoring/wiki/Installing-FFmpeg) on how to get FFMPEG on your machine.
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
To use this package, we'll be assuming you already have the required setup to use Remotion. The dependencies that Remotion requires you to have pre-installed on your machine are Node.js and FFMPEG. You can take a look at this [guide](https://github.com/adaptlearning/adapt_authoring/wiki/Installing-FFmpeg) on how to get FFMPEG on your machine.
```

File: packages/player/README.md [hunk 4]
```diff
@@ -4,5 +4,76 @@
...
+
+The `@remotion/player` package allows you to include a video powered by Remotion in a React application. To gest started with using this package, you need to have this package installed as a dependency in your project. 
+
+The command below does that for you.
+
+```bash
+npm install @remotion/player
+```
+
+Although there are other packagem managers that you can use to get this depedency in your projects. We have [yarn](https://yarnpkg.com/) and [pnpm](https://pnpm.io/). You can go whichever one is most convenient for you. Below are the commands that will get the package into your project.
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
There are other package managers that you can use to get this dependency in your projects. We have [yarn](https://yarnpkg.com/) and [pnpm](https://pnpm.io/). You can go with whichever one is the most convenient for you. Below are the commands that will get the package into your project.
```

File: packages/player/README.md (hunk 1 above)
Comment (anush008) [inline packages/player/README.md]:
```suggestion
- `fps`: The frame rate of the video.
```

File: packages/player/README.md [hunk 5]
```diff
@@ -4,5 +4,76 @@
...
+yarn add @remotion/player
+```
+
+```bash
+pnpm install @remotion/player
+```
+
+## remotion/player in action.
+
+Now that you have this package as a dependency in your React project. It is tie to see some of the basic examples that you can spin up with this package.
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
Now that you have this package as a dependency in your React project. It is time to see some of the basic examples that you can spin up with this package.
```

File: packages/player/README.md [hunk 6]
```diff
@@ -4,5 +4,76 @@
...
+
+The `player` package can be imported as a react component from the library, which you can make use of in your components, either by nesting it in a custom component of yours, or simply making it a standalaone component of its own.
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
The `player` package can be imported as a React component from the library, which you can make use of in your components, either by nesting it in a custom component of yours or simply making it a standalone component.
```

File: packages/player/README.md [hunk 7]
```diff
@@ -4,5 +4,76 @@
...
+      compositionHeight={1080}
+      fps={30}
+    />
+  );
+};
+```
+
+## The props in Player and their roles
+
+- `component`: the component prop takes in a component, a javascript function or a class when it is exported from its module
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
- `component`: the component prop takes in a React component or a function that returns a dynamic import.
```

File: packages/player/README.md [hunk 8]
```diff
@@ -4,5 +4,76 @@
...
+- `durationInFrames`: represents the amount of time a frame will take to render
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
- `durationInFrames`: The duration of the video in frames.
```

File: packages/player/README.md [hunk 9]
```diff
@@ -4,5 +4,76 @@
...
+- `compositionHeight`: represents the height that video component will occupy in the viewport
+- `compositionWidth`: represents the width that video component will occupy in the viewport
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
- `compositionWidth`: The width of the composition in pixels.
```

File: packages/player/README.md [hunk 10]
```diff
@@ -4,5 +4,76 @@
...
+    />
+  );
+};
+```
+
+## The props in Player and their roles
+
+- `component`: the component prop takes in a component, a javascript function or a class when it is exported from its module
+- `durationInFrames`: represents the amount of time a frame will take to render
+- `compositionHeight`: represents the height that video component will occupy in the viewport
```
Comment (anush008) [inline packages/player/README.md]:
```suggestion
- `compositionHeight`: The height of the composition in pixels.
```

**Review by Anush90 — CHANGES_REQUESTED** [review]
Hello @Caleb335 . Thanks for P.R. Greatly appreciated.
There are quite a few typos that need to be corrected. The write-up seems verbose. The props segment can be entirely replaced with a link to the [@remotion/player API](https://www.remotion.dev/docs/player/api) documentation, since there a quite a few props(not included in the write-up).
Hope you'll get these fixed. So that Jonny can approve the changes.

File: packages/player/README.md (hunk 8 above)
Comment (anush008) [reply packages/player/README.md]: The entire props segment can be replaced with a link to the [@remotion/player API](https://www.remotion.dev/docs/player/api) documentation which would make it much more comprehensive.

**Comment by JonnyBurger** [comment]
Thank you both! I’m gone for the weekend but will give this PR attention during next week 😊

**Comment by Caleb335** [comment]
> Hello @Caleb335 . Thanks for P.R. Greatly appreciated.
> 
> > * There are quite a few typos that need to be corrected.
> > * The write-up seems verbose.
> > * The props segment can be entirely replaced with a link to the [@remotion/player API](https://www.remotion.dev/docs/player/api) documentation, since there are a few props not included in the write-up.
> 
> Hope you'll get these fixed so that Jonny can approve the changes.

Thank you so much for your observation, @Anush90. I really appreciate it! 😊 

The typos were a result of me typing too fast. Thank you for the suggested changes that you made. I'll commit them now.

File: packages/player/README.md (hunk 1 above)
Comment (kaf-lamed-beyt) [reply packages/player/README.md]: Thanks, @Anush90

**Comment by Anush90** [comment]
@Caleb335 Can you make it a little more concise? As of now, the write-up seems verbose.
P.S. Take this into consideration.
> * The props segment can be entirely replaced with a link to the [@remotion/player API](https://www.remotion.dev/docs/player/api)

**Comment by Caleb335** [comment]
> @Caleb335 Can you make it a little more concise? As of now, the write-up seems verbose. P.S. Take this into consideration.
> 
> > * The props segment can be entirely replaced with a link to the [@remotion/player API](https://www.remotion.dev/docs/player/api)

Do you mean I should take out the props segment? @Anush90

**Comment by Anush90** [comment]
> Do you mean I should take out the props segment? @Anush90

Yes. There are quite a few props missing.
Adding them to the write-up itself wouldn't be a good idea.
Hence, linking to the [@remotion/player API](https://www.remotion.dev/docs/player/api) will make more comprehensive.

**Comment by Caleb335** [comment]
> > Do you mean I should take out the props segment? @Anush90
> 
> Yes. There are a few props missing. Adding them to the write-up itself wouldn't be a good idea. Hence, linking to the [@remotion/player API](https://www.remotion.dev/docs/player/api) will make it more comprehensive.

Okay. That sounds awesome! 

But, what if we create a table that has all the functions of all the props on the package's page on npm. wouldn't it be better than having people redirected to the docs page?

This is just a suggestion BTW. Kindly let me know what you think.

**Comment by Anush90** [comment]
@Caleb335 Adding up to your suggestion, the current implementation of the props segment would work fine if you add the following line.
> For a complete reference of the available props, refer to [@remotion/player API](https://www.remotion.dev/docs/player/api).

**Comment by Caleb335** [comment]
Awesome! 😎

I'll do that now

**Comment by Caleb335** [comment]
Hi @Anush90...

Do you think I should create a separate PR for the "@remotion/gif" package?

**Comment by Anush90** [comment]
@Caleb335 Yes.

**Comment by Caleb335** [comment]
> @Caleb335 Yes.

Hi @Anush90,

I pushed the changes of the `remotion/gif` readme to this branch accidentally. I hope that is fine

**Review by JonnyBurger — APPROVED** [review]
Thanks for making and iterating over this guys! 🙌 Appreciate your help a lot!

I also added some tweaks, mainly tried to use the correct technical terminology in all cases and also sometimes made sentences shorter without removing information or linking to the documentation. I've learned that the less you write, the more likely people will actually read it lol.

**Comment by Caleb335** [comment]
> I've learned that the less you write, the more likely people will actually read it lol.

Oh wow. This sounds nice though. I'd also try imbibing this pattern.

**Pull request merged**